swatch-contracts/src/test/java/com/redhat/swatch/contract/AwsContractLifecycleIntegrationTest.java

```diff
@@ -0,0 +1,179 @@
+            "cloudIdentifiers": {
+              "seller": {
+                "id": "%s"
+              },
+              "awsCustomerId": "%s",
+              "awsCustomerAccountId": "%s",
+              "productCode": "testProductCode",
+              "partner": "aws_marketplace"
+            }
+          }
+          """,
+          RH_SUBSCRIPTION_NUMBER, AWS_SELLER_ID, AWS_CUSTOMER_ID, AWS_ACCOUNT_ID);
+
+  static String AWS_PARTNER_API_RESPONSE_CONTRACT_CREATED =
+      String.format(
+          """
+          {
+              "content": [
+                  {
+                      "sourcePartner": "aws_marketplace",
+                      "partnerIdentities": {
+                          "awsCustomerId": "%s",
+                          "customerAwsAccountId": "%s",
+                          "sellerAccountId": "%s"
+                      },
+                      "rhAccountId": "%s",
+                      "rhEntitlements": [
+                          {
+                              "sku": "BASILISK",
+                              "subscriptionNumber": "%s"
+                          }
+                      ],
+                      "purchase": {
+                          "vendorProductCode": "%s",
+                          "contracts": [
+                              {
+                                  "startDate": "%s",
+                                  "dimensions": [
+                                      {
+                                          "name": "four_vcpu_hour",
+                                          "value": "730"
+                                      }
+                                  ]
+                              }
+                          ]
+                      },
+                      "status": "STATUS",
+                      "entitlementDates": {
+                          "startDate": "%s"
+                      }
+                  }
+              ],
+              "page": {
+                  "size": 0,
+                  "totalElements": 1,
+                  "totalPages": 0,
+                  "number": 0
+              }
+          }
+          """,
+          AWS_CUSTOMER_ID,
+          AWS_ACCOUNT_ID,
+          AWS_SELLER_ID,
+          RH_ORG_ID,
+          RH_SUBSCRIPTION_NUMBER,
+          AWS_PRODUCT_CODE,
+          UMB_SUBSCRIPTION_START_DATE,
+          UMB_SUBSCRIPTION_START_DATE);
+
+  @BeforeEach
+  @Transactional
+  public void setup() {
+    OfferingEntity offering = new OfferingEntity();
+    offering.setSku("BASILISK");
+    offeringRepository.persist(offering);
+  }
+
+  @AfterEach
+  @Transactional
+  public void cleanup() {
+    subscriptionRepository.deleteAll();
+    contractRepository.deleteAll();
+    offeringRepository.deleteAll();
+  }
+
+  @Test
+  @Transactional
+  void testAwsContractLifecycleHandledAppropriately() throws JsonProcessingException {
+    stubPartnerSubscriptionApi(AWS_PARTNER_API_RESPONSE_CONTRACT_CREATED);
```

Comment:
This isn't really testing the changes I made but I started it so figured I'd check it in.